pkg/components/controllers.go

```diff
@@ -219,7 +221,14 @@ func startDNSController(cfg *config.MicroshiftConfig, kubeconfigPath string) err
 		klog.Warningf("Failed to apply serviceAccount %v %v", sa, err)
 		return err
 	}
-	if err := assets.ApplyConfigMaps(cm, nil, nil, kubeconfigPath); err != nil {
+	// set DNS forward to systemd-resolved resolver if exists
+	// https://github.com/coredns/coredns/blob/master/plugin/loop/README.md#troubleshooting-loops-in-kubernetes-clusters
+	if _, err := os.Stat(config.DefaultSystemdResolvedFile); !errors.Is(err, os.ErrNotExist) {
+		extraParams["UpstreamResolver"] = config.DefaultSystemdResolvedFile
```

Comment:
Checking only for `os.ErrNotExist` seems too specific.  For example, `err` could be a "permission denied" error, in which case it wouldn't make sense to use the path.  Why not just check `err != nil`?

Reply:
I think it tries to ignore other errors and continue to use systemd-resolve file. But I see your point there could be other errors preventing the file be read by microshift. I will use the the generic err check.

Reply:
Fixed.